src/EditDialog.cpp

```diff
@@ -94,38 +94,44 @@ void EditDialog::loadData(const QByteArray& data)
     case Null:
         switch (editMode) {
         case TextEditor:
+            // The text widget buffer is now the main data source
+            dataSource = TextBuffer;
+
             // Empty the text editor contents, then enable text editing
             ui->editorText->clear();
             ui->editorText->setEnabled(true);
```

Comment:
This caused the error, as it indirectly calls the `editTextChanged` function, but changed all of them as it's safer.